### PR TITLE
Add draft of documentation

### PR DESCRIPTION
**Pull request by rgaiacs**
Use Sphinx as documentation framework.

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Installation
```
Comment (ChasNelson1990) [inline docs/installation.rst:1]: This a separate point but should we have develop install instructions on the github somewhere?

**Review by ChasNelson1990 — APPROVED** [review]
Looks good to me!

**Review by ChasNelson1990 — APPROVED** [review]

File: docs/installation.rst (hunk 1 above)
Comment (tinosulzer) [reply docs/installation.rst]: Yes, I guess they can be here too, and the readme should point to this

Comment (rgaiacs) [reply docs/installation.rst]: I added to the documentation.

**Comment by tinosulzer** [comment]
To fix the flake8 tests, you can just install and run the `black` package

File: docs/examples/minimal-class.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from r2t2 import add_reference
```
Comment (de-code) [inline docs/examples/minimal-class.py:1]: minor suggestion: maybe it would be good to keep the underscore (`_`) for Python files?

**Comment by de-code** [comment]
Not sure how to add a comment to a renamed file. `docs/examples/reference-annotation.py` is probably superseded by `minimal.py`

File: docs/examples/minimal-class.py (hunk 2 above)
Comment (rgaiacs) [reply docs/examples/minimal-class.py:1]: The `...` will fail in any test and when executing. This is why I removed the `...`.

**Comment by tinosulzer** [comment]
Should `docs/requirements.txt` be removed? It seems to duplicate the `setup.py`

**Comment by rgaiacs** [comment]
`docs/requirements.txt` is a place holder to be used by Read The Docs and other users building the documentation. At the moment it doesn't install anything.

**Comment by tinosulzer** [comment]
I have just noticed the installation file is duplicated: once for the pip install, once for cloning and installing locally

**Comment by rgaiacs** [comment]
@tinosulzer one installation instructions is for users and the other is for developers. They are in different folders. And use in different places.

**Comment by tinosulzer** [comment]
Ah yeah I see, the sphinx format is confusing with subtitles larger than titles
Screen Shot 2020-09-02 at 2 12 48 PM

**Comment by jezcope** [comment]
@all-contributors please add @rgaiacs for doc

Thanks Raniere!

**Comment by jezcope** [comment]
@all-contributors please add @rgaiacs for examples

File: docs/examples/minimal-class.py (hunk 2 above)
Comment (de-code) [reply docs/examples/minimal-class.py:1]: I would just not use `...` or convert it to a comment (e.g. `# ...`).

**Pull request merged**